packages/@glimmer/application/src/helpers/user-helper.ts

```diff
@@ -26,10 +26,11 @@ export default function buildUserHelper(helperFunc): GlimmerHelper {
 export class SimplePathReference<T> implements PathReference<T> {
```

Comment:
Can we just use a ConstReference for wrapping user helpers instead of reimplementing PathReference?

Reply:
This seems to indicate we need a `PathReference` implementation. https://github.com/glimmerjs/glimmer-vm/blob/master/packages/%40glimmer/runtime/lib/environment.ts#L312. I agree this can be better but I think we should land this work first and then do all the type fixes as a fast follow. The fact that these things are volatile is much worst then the duplication.

Reply:
I think there is an PathReference that is const.

Reply:
PrimitiveReference extends ConstReference and implements PathReference, and ValueReference extends PrimitiveReference